### PR TITLE
427 : [Refactoring] Extract Logout service model from Logout controller

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR is regarding logout controller in which logout logic is moved to logout service model

### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#427: [Refactoring] Extract Logout service model from Logout controller

### CODE REVIEW

**Comment by konarshankar07** [comment]
Please don't merge this PR. I'm still working on Unit test 
Thanks

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — DISMISSED** [review]

**Comment by konarshankar07** [comment]
Hello @sivaschenko ,
Please don't merge this PR yet. As coveralls still showing approx 89%, I'm working on this issue and update you soon.
image
Thanks

**Review by sivaschenko — APPROVED** [review]

File: AdobeIms/Test/Unit/Model/LogOutTest.php [hunk 1]
```diff
@@ -97,6 +97,9 @@ public function testExecute(): void
     {
         $this->userContextInterfaceMock->expects($this->exactly(1))
             ->method('getUserId')->willReturn(1);
+        $this->userProfileInterfaceMock->expects($this->once())
+            ->method('getAccessToken')
```
Comment (konarshankar07) [inline AdobeIms/Test/Unit/Model/LogOutTest.php:101]: Hello @sivaschenko ,
Why we need to test the `getAccessToken` method ? As it's not present in `AdobeIms/Model/LogOut.php`. Sorry, Correct me If I'm wrong.
Thanks

Comment (sivaschenko) [reply AdobeIms/Test/Unit/Model/LogOutTest.php:101]: Hi @konarshankar07 , this method is called in `AdobeIms/Model/LogOut.php` to retrieve a token for Logout URL generation

Comment (konarshankar07) [reply AdobeIms/Test/Unit/Model/LogOutTest.php:101]: Yes, You are right. Thanks for the clarification :)

**Pull request merged**